test/integration/auth.test.js

```diff
@@ -254,4 +256,296 @@ describe("auth", function () {
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUrl);
+  });
+
+  it("should redirect the google user to realdevsquad.com if non RDS URL provided, any url that is other than *.realdevsquad.com is invalid", async function () {
+    await addUserToDBForTest(googleUserInfo[1]);
+    const invalidRedirectUrl = new URL("https://google.com").href;
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl")).href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get(`/auth/google/callback`)
+      .query({ code: "codeReturnedByGoogle", state: invalidRedirectUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUiUrl);
+  });
+
+  it("should redirect the google user to realdevsquad.com if invalid redirect URL provided", async function () {
+    await addUserToDBForTest(googleUserInfo[1]);
+    const invalidRedirectUrl = "invalidURL";
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl")).href;
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    const res = await chai
+      .request(app)
+      .get(`/auth/google/callback`)
+      .query({ code: "codeReturnedByGoogle", state: invalidRedirectUrl })
+      .redirects(0);
+    expect(res).to.have.status(302);
+    expect(res.headers.location).to.equal(rdsUiUrl);
+  });
+
+  it("should send a cookie with JWT in the response for google user", function (done) {
+    const rdsUiUrl = new URL(config.get("services.rdsUi.baseUrl"));
+
+    sinon.stub(passport, "authenticate").callsFake((strategy, options, callback) => {
+      callback(null, "accessToken", googleUserInfo[0]);
+      return (req, res, next) => {};
+    });
+
+    chai
+      .request(app)
+      .get("/auth/google/callback")
+      .query({ code: "codeReturnedByGoogle" })
+      .redirects(0)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        expect(res).to.have.status(302);
+        expect(res.headers["set-cookie"]).to.have.length(1);
+        expect(res.headers["set-cookie"][0])
+          .to.be.a("string")
+          .and.satisfy((msg) => msg.startsWith(config.get("userToken.cookieName")));
+        expect(res.headers["set-cookie"][0]).to.include("HttpOnly");
+        expect(res.headers["set-cookie"][0]).to.include("Secure");
+        expect(res.headers["set-cookie"][0]).to.include(`Domain=${rdsUiUrl.hostname}`);
+        expect(res.headers["set-cookie"][0]).to.include("SameSite=Lax");
+
+        return done();
+      });
+  });
+
+  it("should redirect the google user to login page if the user is a developer", async function () {
```

Comment:
How do we know if its developer without users already being present in RDS

Reply:
If the user is not already present in the RDS, we treat them as a new user and create an account for them. In this case, we don’t need to check for the "developer" role, since the user doesn't exist yet. The role (e.g., "developer") will be assigned during the new-signup application (in Future).